src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets

```diff
@@ -13,6 +13,21 @@ This file contains the .NET 5-specific targets to customize ILLink
       Condition=" '$(PublishTrimmed)' == 'true' "
       AfterTargets="ComputeResolvedFilesToPublishList"
       DependsOnTargets="GetReferenceAssemblyPaths">
+    <ItemGroup Condition=" '$(AndroidLinkMode)' == 'Full' ">
+      <!-- Mark all assemblies to be linked if we're linking all assemblies -->
+      <ResolvedFileToPublish
+          Update="@(ResolvedFileToPublish)"
+          Condition=" '%(ResolvedFileToPublish.Extension)' == '.dll' and '%(ResolvedFileToPublish.Filename)%(ResolvedFileToPublish.Extension)' != '$(TargetFileName)' and '%(ResolvedFileToPublish.AssetType)' != 'native' "
+          TrimMode="link"
+      />
+      <!-- Linking $(TargetFileName) causes: error IL1020: No files to link were specified. Use one of '-a|-r|-x' options -->
+      <ResolvedFileToPublish
+          Update="@(ResolvedFileToPublish)"
+          Condition=" '%(ResolvedFileToPublish.Filename)%(ResolvedFileToPublish.Extension)' == '$(TargetFileName)' "
+          IsTrimmable="false"
+          TrimMode=""
+      />
+    </ItemGroup>
```

Comment:
@rolfbjarne @radekdoulik I had to exclude the main assembly for `AndroidLinkMode=full` or I was hitting:
```
error IL1020: No files to link were specified. Use one of '-a|-r|-x' options
```
I think it is saying there is no "root assembly", so the error message isn't quite right?

Reply:
Yes, the error message isn't quite right.

This is what I did: https://github.com/xamarin/xamarin-macios/blob/e72ef18bebad34e4c26500bfed5d50410437fb80/dotnet/targets/Xamarin.Shared.Sdk.targets#L216-L217

Reply:
Yes, `TrimmerRootAssembly ` seems to be the way to specify the [root assembly](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options#root-assemblies).